src/backend/core/models/workflow_metrics_models.py

```
@@ -0,0 +1,362 @@
+"""
+Pydantic models for workflow metrics.
+
+These models provide:
+1. Shared base classes to eliminate duplication
+2. Built-in validation via Pydantic
+3. Backward compatibility via model_config (extra='ignore' removes obsolete fields automatically)
+4. Clean serialization/deserialization
+"""
+
+from datetime import datetime
+from typing import Dict, List, Optional, Any
```

Ruff (0.14.10)

`typing.Dict` is deprecated, use `dict` instead

(UP035)